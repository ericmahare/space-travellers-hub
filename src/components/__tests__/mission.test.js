import renderer from 'react-test-renderer';
import { screen, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Missions from '../components/mission/missionTable';

const StoreProvider = () => (
  <Provider store={store}>
    <Missions />
  </Provider>
);

describe('mission component test', () => {
  it('renders correctly', () => {
    const RenderComponent = renderer.create(<StoreProvider />).toJSON();
    expect(RenderComponent).toMatchSnapshot();
  });

  it('renders data from api', async () => {
    render(<StoreProvider />);
    const joinBtns = await screen.findAllByText(/Join Mission/i);
    expect(joinBtns).toHaveLength(10);
  });
});