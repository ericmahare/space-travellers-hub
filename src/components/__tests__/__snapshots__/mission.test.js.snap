// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mission component test renders correctly 1`] = `
<table
  className="table"
>
  <tbody>
    <tr>
      <th>
        Mission
      </th>
      <th>
        Description
      </th>
      <th>
        Status
      </th>
      <th>
         
      </th>
    </tr>
  </tbody>
</table>
`;